src/scriptdev2/scripts/kalimdor/durotar.cpp

```diff
@@ -130,11 +225,12 @@ bool EffectDummyCreature_lazy_peon_awake(Unit* pCaster, uint32 uiSpellId, SpellE
     // always check spellid and effectindex
     if (uiSpellId == SPELL_AWAKEN_PEON && uiEffIndex == EFFECT_INDEX_0)
     {
+        //TODO some of these checks are now redundant, should be removed by someone who knows how to do so without breaking things
```

Comment:
Don't doubt yourself.

ask yourself if the code below makes sense; basically what's happening is the spell is cast on the peon; if the spell failed in checkcast, it won't get to hit the peon, so all of this won't execute at all. but what you are really checking, is if the target is not sleeping, or it's not a player casting it, or the target isn't a peon, the.... return true? - that's all wrong. It should be return false in that case.

the reason why it was used in a weird way before is because the spell was used by the peon on himself in the old script. It isn't any longer. the check is completely redundant, especially when it returns the absolute opposite of what it ought to.